### PR TITLE
feat(client): validate constructor

### LINKED ISSUE #889: Runtime validation for Prisma client constructor parameters
## Problem
If there is a typo/misuse in Prisma client's constructor parameters. We silently get a Prisma client object. 

## Suggested solution
It should fail with a runtime validation error like queries fail. 

## Additional context
I ran into this multiple time when using Prisma client in Javascript where I had either a typo or I was inadvertently using a param that didn't exist.

### PR DESCRIPTION
**Pull request by timsuchanek**
Closes https://github.com/prisma/prisma-client-js/issues/889

### CODE REVIEW

**Comment by Jolg42** [comment]
Looks like Studio tests are failing https://github.com/prisma/prisma/pull/4176/checks?check_run_id=1364713203#step:6:592
```
 Generating Prisma Client ... done
  console.error
    Unable to start Prisma Client:  PrismaClientValidationError: Unknown property engine provided to PrismaClient constructor.
        at validatePrismaClientOptions (/home/runner/.cache/prisma/studio/home-runner-work-prisma-prisma-src-packages-cli-src-__tests__-fixtures-studio-test-project/runtime/index.js:77932:13)
        at new NewPrismaClient (/home/runner/.cache/prisma/studio/home-runner-work-prisma-prisma-src-packages-cli-src-__tests__-fixtures-studio-test-project/runtime/index.js:77996:9)
        at Photon.start (/home/runner/work/prisma/prisma/src/node_modules/.pnpm/@prisma/studio-transports@0.308.0/node_modules/@prisma/studio-transports/build/photon.js:92:33)
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
        at PhotonService.respond (/home/runner/work/prisma/prisma/src/node_modules/.pnpm/@prisma/studio-server@0.308.0/node_modules/@prisma/studio-server/build/services/PhotonService.js:88:25)
        at Object.StudioServer.onWSMessage [as callback] (/home/runner/work/prisma/prisma/src/node_modules/.pnpm/@prisma/studio-server@0.308.0/node_modules/@prisma/studio-server/build/StudioServer.js:79:35)

      at Photon.start (../../node_modules/.pnpm/@prisma/studio-transports@0.308.0/node_modules/@prisma/studio-transports/build/photon.js:132:25)
          at runMicrotasks (<anonymous>)
      at PhotonService.respond (../../node_modules/.pnpm/@prisma/studio-server@0.308.0/node_modules/@prisma/studio-server/build/services/PhotonService.js:88:25)
      at Object.StudioServer.onWSMessage [as callback] (../../node_modules/.pnpm/@prisma/studio-server@0.308.0/node_modules/@prisma/studio-server/build/StudioServer.js:79:35)
```

**Comment by timsuchanek** [comment]
@Jolg42 yep. @madebysid and I are already debugging

File: src/packages/client/src/__tests__/validatePrismaClientOptions.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    )
+  })
+})
+
+describe('invalid options', () => {
+  test('typos', () => {
+    expect(() =>
+      validatePrismaClientOptions(
+        {
+          errorrFormat: 'minimal',
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/validatePrismaClientOptions.test.ts:48]: Trying to mess with some brains? 😄

File: src/packages/client/src/runtime/utils/validatePrismaClientOptions.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          const didYouMean = getDidYouMean(level, logLevels)
+          throw new PrismaClientValidationError(`Invalid log level "${level}" provided to PrismaClient constructor.${didYouMean}`)
+        }
+      }
+    }
+
+    for (const option of options) {
+      validateLogLevel(option)
+
+      const validators = {
```
Comment (Jolg42) [inline src/packages/client/src/runtime/utils/validatePrismaClientOptions.ts:81]: Could maybe be renamed to `logValidators` as there is already a top level validators const?

**Review by Jolg42 — APPROVED** [review]
Nice! Maybe the logs could link to https://www.prisma.io/docs/concepts/components/prisma-client/constructor ?

**Comment by Jolg42** [comment]
Note From what I understand it fails at the first error found ✅

**Comment by Jolg42** [comment]
✨ Nice!

**Pull request merged**